File: ahammer3-cs50-2018-x-cash-20180925T013829Z/ahammer3-cs50-2018-x-cash-20180925T013829Z/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float change;
    int n;  //store change value
    int counter;

    while (true)
    {
        printf("Change owed in dollars: \n"); //get input of change owed in float value
        change = get_float();

        if (change > 0)
        {
            break;
        }

    }

    n = round(change * 100);  //convert decimal to int and round to avoid flooring

    counter = 0;  //count the number of coins

    while (n >= 25)  //no quarters
    {
        n = n - 25;
        counter += 1;
    }

    while (n >= 10)  //no dimes
    {
        n = n - 10;
        counter += 1;
    }

    while (n >= 5)  //no nickels
    {
        n = n - 5;
        counter += 1;
    }

    printf("%d\n", counter + n); //prints the remainder n as pennies in 0-4 range

    return 0;
}

File: ahammer3-cs50-2018-x-sentimental-cash-20181107T192438Z/ahammer3-cs50-2018-x-sentimental-cash-20181107T192438Z/cash.py
# This program outputs the minimum # coins required to make the change given

# import modules
import cs50
import math

# get the change owed in a loop that runs at least once
while True:
    dollars = cs50.get_float("Change owed in dollars: ")
    cents = math.floor(dollars * 100)

    if cents > 0:
        break

# calculate each type of coin using mods
quarters = cents // 25
dimes = (cents % 25) // 10
nickels = ((cents % 25) % 10) // 5
pennies = ((cents % 25) % 10) % 5

# simply print it out!
print(f"{quarters + dimes + nickels + pennies}")